crates/driver/src/domain/time.rs

```diff
@@ -0,0 +1,40 @@
+use {
+    crate::{
+        infra::{self},
+        util,
+    },
+    thiserror::Error,
+};
+
+/// A datetime representing a deadline until something needs to be done.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Deadline(chrono::DateTime<chrono::Utc>);
```

Comment:
I think deadline should take the timeouts and the target deadline and then expose methods that give us the individual sub-deadlines for the solver and for completing the merging. This way we don't have to leak all of the time math into the different sub-components and can just pass a single Deadline instance around.

Reply:
Fixed.